doc/examples/xx_applications/plot_haar_extraction_selection_classification.py

```diff
@@ -130,10 +128,13 @@ def extract_feature_image(img, feature_type, feature_coord=None):
 # Note: we could select those features from the
 # original matrix X but we would like to emphasize the usage of `feature_coord`
 # and `feature_type` to recompute a subset of desired features.
-X = np.array(Parallel(n_jobs=-1)(
-    delayed(extract_feature_image)(img, selected_feature_type,
-                                   selected_feature_coord)
-    for img in images))
+
+# Delayed the computation and build the graph using dask
+X = delayed(extract_feature_image(img, selected_feature_type,
```

Comment:
@glemaitre I think here you want instead:

```python
Xd = [delayed(extract_feature_image)(img, selected_feature_type, selected_feature_coord)
      for img in images]
X = np.array(dask.compute(*Xd))
```

Then you won't need dask.bag.

By the way, do we really need multiprocessing here? Aren't most operations gil-released C/Cython?

Reply:
Basically I don't need dask.bag. I need cloudpickle because it is imported when importing multiprocessing from dask. To avoid installing dask[complete] since I only need cloudpickle, I installed dask[array,bag] which install dask, numpy, and cloudpickle

Regarding the syntax, I think that they are equivalent.

Reply:
This is fine; another option would be to add cloudpickle to the requirements explicitly with a comment.